### DISCUSSION TITLE
Can I add to a delayed transitions a guard?

### DISCUSSION THREAD

**Discussion by sisrugo92** [statelyai/xstate#2288] [Q&A]
Hi, 
Can I add a guard to a delayed transition?

for example, I want the green state `after` to happen only if some flag is true.

```ts
const lightDelayMachine = createMachine({
  id: 'lightDelay',
  initial: 'green',
  states: {
    green: {
      after: {
        // after 1 second, transition to yellow
        1000: { target: 'yellow' }
      }
    },
    yellow: {
      after: {
        // after 0.5 seconds, transition to red
        500: { target: 'red' }
      }
    },
    red: {
      after: {
        // after 2 seconds, transition to green
        2000: { target: 'green' }
      }
    }
  }
});
```

Thank you , Si

**Comment by davidkpiano** [comment]
Sure, you would add a `cond` to your transition, just like any other transition:

```js
    green: {
      after: {
        // after 1 second, transition to yellow based on some condition
        1000: { target: 'yellow', cond: 'someCondition' }
      }
    },
```

**Reply by sisrugo92** [reply]
The condition will be checked before the delay?

**Reply by johtso** [reply]
The condition is applied to the action triggered by the delayed event, so no.

The docs describe what this syntax translates to, so you could write the sending of your conditional delayed event based on that: https://xstate.js.org/docs/guides/delays.html#behind-the-scenes

I guess there is a question of whether this is a common enough use case that it should somehow be added to the delay syntax.. not sure if there is a way that could elegantly be achieved though..
